rpcs3/Emu/RSX/Program/program_util.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "util/types.hpp"
+#include "../gcm_enums.h"
+
+#include <concepts>
+
+namespace rsx
+{
+#pragma pack(push, 1)

```

Comment:
What's this?

Reply:
Force byte 1 alignment in structures. Usually the compiler is allowed to insert padding bytes, but this payload is getting copied to VRAM and must obey GPU packing method.

Reply:
It actually does nothing in the structures used here because members are perfectly aligned as is.

Reply:
It's missing a comment then anyway

Reply:
It's a fairly standard method of bit-packing, it's basically MSVCs version of __attribute__((packed)). I'll add a comment for clarity though.
As for its inclusion, I did so because these structs will be rewritten in the future to use color transformation payload structs and avoid all the branching we currently have on the shader cores.